### PR TITLE
[refactor] annotator structure reorganization

### PR DESCRIPTION
**Pull request by subdavis**
* moves layers to subfolder
* moves controls to subfolder
* moves annotators to subfolder
* extracts common annotator logic to mixin
* documents annotation app design choices
* documents annotator app API
* Adds `yarn.lock` because somehow that was never committed. 
* Removes some chainWebpack config that 1) didn't make sense and 2) seemed to be breaking my build.  We really need to document when we do non-standard things....
* Uses a proper release version of GWC otherwise the dependency will change every time master updates.

This is just a first-pass reorganization.  `Viewer.vue` will need more consideration.  These were just some of the most obvious areas for improvement.

### CODE REVIEW

File: client/vue.config.js [hunk 1]
```diff
@@ -12,14 +12,6 @@ module.exports = {
...
   chainWebpack: config => {
     config.resolve.symlinks(false);
-    config.module
-      .rule("additional")
-      .test(/\.m?jsx?$/)
-      .include.add(/@girder\/components/)
-      .add(/vue-utilities/)
-      .end()
-      .use()
-      .loader("babel-loader");
```
Comment (subdavis) [inline client/vue.config.js:22]: This broke my build.  I have no idea what it was here for.  I definitely don't know why JSX is being referenced.

**Comment by subdavis** [comment]
Screenshot from 2020-04-03 16-08-16

This is what happens, btw.

File: client/src/components/annotators/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## state
+
+The state available on the injected Vue annotator instance.
+
+```js
+viewer // geojs viewer
+playing // boolean
+frame // number
+maxFrame // number
```
Comment (BryonLewis) [inline client/src/components/annotators/README.md:13]: viewer - is there a better name we should be calling this object?  imageViewer, frameViewer, imageRenderer, frameRenderer or something like that.  Looking at the code my mind immediately goes 'there's a Viewer.vue, must have some relationship to that' and that turns out to be wrong.  Could be better to rename the Viewer.vue to something more fitting.

Comment (subdavis) [reply client/src/components/annotators/README.md:13]: It's an instance of `geo.map()` so something like `geoViewer` or `geoMap` I think would be appropriate.

File: client/src/components/annotators/VideoAnnotator.vue [hunk 3]
```diff
@@ -65,38 +33,10 @@ export default {
...
-      ];
-      interactorOpts.zoomAnimation = {
-        enabled: false
-      };
-      interactorOpts.momentum = {
-        enabled: false
-      };
-      interactorOpts.wheelScaleY = 0.2;
-      this.viewer.interactor().options(interactorOpts);
+      this.baseInit();
```
Comment (AlmightyYakob) [inline client/src/components/annotators/VideoAnnotator.vue:39]: Can you add a comment here referencing the mixin?

File: client/src/components/annotators/annotator.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    this.provided.$on("prev-frame", this.prevFrame);
+    this.provided.$on("next-frame", this.nextFrame);
+    this.provided.$on("play", this.play);
+    this.provided.$on("pause", this.pause);
+    this.provided.$on("seek", this.seek);
+    this.emitFrame();
+    this.emitFrame = throttle(this.emitFrame, 200);
+  },
+  methods: {
+    baseInit() {
```
Comment (AlmightyYakob) [inline client/src/components/annotators/annotator.js:46]: Might be good to add a comment to this function as well explaining what it's doing, if possible.

Comment (subdavis) [reply client/src/components/annotators/annotator.js:46]: this is explained in `src/components/annotators/README.md`

**Pull request merged**